src/pages/ThorChainLP/Pool/components/PoolChart.tsx

```diff
@@ -1,34 +1,131 @@
 import { Button, ButtonGroup, Center, Flex, Stack } from '@chakra-ui/react'
-import { FaChartSimple } from 'react-icons/fa6'
-import { useTranslate } from 'react-polyglot'
-import { RawText } from 'components/Text'
+import { useQuery } from '@tanstack/react-query'
+import type { SingleValueData, UTCTimestamp } from 'lightweight-charts'
+import { useCallback, useMemo, useState } from 'react'
+import { reactQueries } from 'react-queries'
+import type { Interval } from 'react-queries/queries/midgard'
+import { SimpleChart } from 'components/SimpleChart'
+import { fromThorBaseUnit } from 'lib/utils/thorchain'
+import type {
+  MidgardSwapHistoryResponse,
+  MidgardTvlHistoryResponse,
+} from 'lib/utils/thorchain/lp/types'
+import { selectUserCurrencyToUsdRate } from 'state/slices/selectors'
+import { store } from 'state/store'
+
+const swapHistoryToChartData = (swapHistory: MidgardSwapHistoryResponse): SingleValueData[] => {
+  if (!swapHistory) return []
+
+  const userCurrencyToUsdRate = selectUserCurrencyToUsdRate(store.getState())
+
+  return swapHistory.intervals.map(interval => {
+    const intervalVolumeFiatUserCurrency = fromThorBaseUnit(interval.totalVolume)
+      .times(interval.runePriceUSD)
+      .times(userCurrencyToUsdRate)
+
+    return {
+      time: Number(interval.startTime) as UTCTimestamp,
+      value: intervalVolumeFiatUserCurrency.toNumber(),
+    }
+  })
+}
+
+const tvlToChartData = (
+  tvl: MidgardTvlHistoryResponse,
+  thorchainNotationAssetId: string,
+): SingleValueData[] =>
+  tvl.intervals.map(interval => {
+    const userCurrencyToUsdRate = selectUserCurrencyToUsdRate(store.getState())
+    const poolDepth = interval.poolsDepth.find(pool => pool.pool === thorchainNotationAssetId)
+    const poolTotalDepth = poolDepth?.totalDepth ?? '0'
+
+    const tvlFiat = fromThorBaseUnit(poolTotalDepth)
+      .times(interval.runePriceUSD)
+      .times(userCurrencyToUsdRate)
+
+    return {
+      time: Number(interval.startTime) as UTCTimestamp,
+      value: tvlFiat.toNumber(),
+    }
+  })
+
+// @ts-ignore we can't make this a partial record as we need to use this as a tuple to spread as useQuery params
+const INTERVAL_PARAMS_BY_INTERVAL: Record<Interval | 'all', [Interval, number]> = {
```

Comment:
nitpick - add `ChartInterval` type with selectable entires and remove the `ts-ignore`

Reply:
ae64488c83